src/Table/Table.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect } from 'react';
+import { useTable, useSortBy, useExpanded, TableState, Column } from 'react-table';
+import { ReactComponent as ArrowUp } from './assets/arrow_drop_up.svg';
+import { ReactComponent as ArrowDown } from './assets/arrow_drop_down.svg';
+import { TableStyle } from './tableStyles';
+
+interface SortByOptions {
+    id: string;
+    desc: boolean;
+}
+
+export interface TableProps<T> {
```

Comment:
For the sake of my understanding... why is this parameterized?  Is it so that it automatically picks up the options that `useTable` expects?  If so, why not declare them explicitly?

Reply:
@adorack-tri  - This was due to my review comment. Initially I mentioned it as a suggestion since we were using `any` everywhere, but after seeing that `react-table` uses a lot of  `object` typing (which is fairly wide open), it doesn't much matter if we explicitly set types or not. But I'm also not for users putting in `TableProps<any>`. So I think the fact that you're asking indicates we still need some better clarity. Here is my suggestion (though I'm open to other thoughts):

@NehalBhandariSlalom - Sorry for leading you this way. I think the best option would be to remove the generic parameter (e.g. remove `T`), and then just make the items that use `T` be simply `object`. I'm not sure we can make it any more specific than that.

Reply:
I'm not that familiar with `react-table`... my comment was because it looked to me like `rowProps` and `options` would have a known structure.  If that's not the case, but it's meaningful to have type info because users might be able to impose some structure, I'm not objecting to it.  Perhaps better docs would help?

Reply:
Yeah, I'm not that familiar with `react-table` either. Yes, better docs would help. Honestly just pointing users to the [react-table useTable documentation](https://react-table.tanstack.com/docs/api/useTable) would help.

Reply:
There has been lot of confusion with the types and again the react-table doesn't have specific types for options and rowProps. I have removed generic <T> type and provided the types that is relevant and won't cause any issues because its also getting used in Lakehouse. Hope it helps now.

Reply:
Also Added the link for react table as suggested by Jason in the documentation.